### PR TITLE
Update bernstein polynomial

### PR DESCRIPTION
**Pull request by drwells**
The current code is fine, but we can clean things up a bit by using a boost implementation (this is also slightly faster).

### CODE REVIEW

File: source/base/polynomials_bernstein.cc [hunk 1]
```diff
@@ -38,11 +17,13 @@ namespace Binomial
     AssertIndexRange(k, n+1);
     std::vector<number> coeff(n + 1, number(0.0));
     for (unsigned int i = k; i < n + 1; ++i)
-      coeff[i] = (pow(number(-1), number(i - k)) * binomial(n, i)
-                  * binomial(i, k));
+      {
+        coeff[i] = ((i - k) % 2 == 0 ? 1 : -1)
+                   * boost::math::binomial_coefficient<number>(n, i)
+                   * boost::math::binomial_coefficient<number>(i, k);
```
Comment (tamiko) [inline source/base/polynomials_bernstein.cc]: Have you checked with the bundled version that ```boost::math``` is available?

Comment (bangerth) [inline source/base/polynomials_bernstein.cc]: We do have this header file in bundled/ indeed.

Comment (tamiko) [inline source/base/polynomials_bernstein.cc]: > We do have this header file in bundled/ indeed.

Let me rephrase. My system has 18 libboost_math_* support libraries
installed. But, yes, you're right. From their contents (I just checked
the boost sources) they're very likely not needed for our use case.

Comment (bangerth) [inline source/base/polynomials_bernstein.cc]: > Let me rephrase. My system has 18 libboost_math_* support libraries installed.
> But, yes, you're right. From their contents (I just checked the boost sources)
> they're very likely not needed for our use case.

I did check that the function he's calling is indeed present in bundled/.

Comment (tamiko) [inline source/base/polynomials_bernstein.cc]: Nvm. A misunderstanding.

Comment (bangerth) [inline source/base/polynomials_bernstein.cc]: For the record, I just compiled everything with the bundled boost and it continues to compile fine.
